wrapanapi/systems/virtualcenter.py

```diff
@@ -225,9 +251,22 @@ def _pick_datastore(self, allowed_datastores):
             key=lambda ds: float(ds.summary.freeSpace) / float(ds.summary.capacity),
             reverse=True)
         if not possible_datastores:
-            raise Exception("No possible datastores!")
+            raise Exception("No datastores are not available for provision, check status of "
+                            "datastore(s)")
         return possible_datastores[0]
 
+    def _pick_datastore_cluster(self):
```

Comment:
Seems this may be a useful method to expose as `pick_datastore_cluster`, so callers can use it directly if they want.